sdk/common/src/main/java/io/opentelemetry/sdk/internal/ExtendedAttributesMap.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.internal;
+
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.incubator.common.ExtendedAttributeKey;
+import io.opentelemetry.api.incubator.common.ExtendedAttributes;
+import io.opentelemetry.api.incubator.common.ExtendedAttributesBuilder;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.BiConsumer;
+import javax.annotation.Nullable;
+
+/**
+ * A map with a fixed capacity that drops attributes when the map gets full, and which truncates
+ * string and array string attribute values to the {@link #lengthLimit}.
+ *
+ * <p>{@link ExtendedAttributes} analog of {@link AttributesMap}.
+ *
+ * <p>This class is internal and is hence not for public use. Its APIs are unstable and can change
+ * at any time.
+ */
+public final class ExtendedAttributesMap extends HashMap<ExtendedAttributeKey<?>, Object>
+    implements ExtendedAttributes {
+
+  private static final long serialVersionUID = -2674974862318200501L;
+
+  private final long capacity;
+  private final int lengthLimit;
+  private int totalAddedValues = 0;
+
+  private ExtendedAttributesMap(long capacity, int lengthLimit) {
+    this.capacity = capacity;
+    this.lengthLimit = lengthLimit;
+  }
+
+  /**
+   * Create an instance.
+   *
+   * @param capacity the max number of extended attribute entries
+   * @param lengthLimit the maximum length of string attributes
+   */
+  public static ExtendedAttributesMap create(long capacity, int lengthLimit) {
```

Comment:
Isn't there some spec mandated default we can use (with an overload) for lengthLimit?